### PR TITLE
Fix/join concurrent

### LINKED ISSUE #1130: How does join handle the case where the client does not have any input?
there is a case of not having any input:

```shell
case="no any data"
arg=$(cat <<-EOF
EOF
)
actual=$(testgrpc ChannelJoin <<<"$arg" 2>&1 || flag=true)
echo $actual
```

this case will output an error:
```text
ERROR:
  Code: Internal
  Message: EOF
```

Do we want it to report an error or not report an error but not output anything?

### PR DESCRIPTION
**Pull request by lianxmfor**
close #1130

### CODE REVIEW

File: oomagent/test/test_channel_join.sh [hunk 1]
```diff
@@ -104,9 +104,43 @@ EOF
...
+
+no_data() {
+   case="no data"
+  arg=$(cat <<-EOF
+EOF
+)
+  actual=$(testgrpc ChannelJoin <<<"$arg" 2>&1 || flag=true)
+  expected=$(cat <<-EOF
+ERROR:
+  Code: Internal
```
Comment (YikSanChan) [inline oomagent/test/test_channel_join.sh:132]: should be INVALID_ARGUMENT instead?

File: oomagent/test/test_channel_join.sh [hunk 2]
```diff
@@ -104,9 +104,43 @@ EOF
...
+EOF
+)
+  assert_eq "$case" "$expected" "$actual"
+}
+
 main() {
   successful_cases
   failed_cases
+  no_entity_rows
+  no_data
```
Comment (jinghancc) [inline oomagent/test/test_channel_join.sh:143]: you could wrap `no_data` and `no_entity_rows` to `failed_cases`

File: oomagent/test/test_channel_join.sh [hunk 3]
```diff
@@ -104,9 +104,43 @@ EOF
...
+)
+  actual=$(testgrpc ChannelJoin <<<"$arg" 2>&1 || flag=true)
+  expected=$(cat <<-EOF
+EOF
+)
+  assert_eq "$case" "$expected" "$actual"
+}
+
+no_data() {
+   case="no data"
```
Comment (jinghancc) [inline oomagent/test/test_channel_join.sh:125]: suggestion: invalid_request

File: oomagent/test/test_channel_join.sh (hunk 1 above)
Comment (lianxmfor) [reply oomagent/test/test_channel_join.sh]: sounds good! done.

File: oomagent/test/test_channel_join.sh (hunk 2 above)
Comment (lianxmfor) [reply oomagent/test/test_channel_join.sh]: done

File: oomagent/test/test_channel_join.sh (hunk 3 above)
Comment (lianxmfor) [reply oomagent/test/test_channel_join.sh]: good suggestion! done

**Review by YikSanChan — APPROVED** [review]
lgtm

**Pull request merged**